### PR TITLE
Add VIA support to DZ60

### PR DESCRIPTION
**Pull request by minajevs**
## Description
Recreated this PR with clean commit state. Previous PR: #7810 

Adds VIA support keymap to `dz60` board. Tested on my current DZ60 build and it works perfect. 

[Related VIA PR](https://github.com/the-via/keyboards/pull/19)

## Types of Changes
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR
* N/A

## Checklist
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/dz60/config.h [hunk 1]
```diff
@@ -4,7 +4,7 @@
 #include "config_common.h"
 
 /* USB Device descriptor parameter */
-#define VENDOR_ID       0xFEED
+#define VENDOR_ID       0x4B46  // "KF" as if KbdFans
 #define PRODUCT_ID      0x2260
```
Comment (Wilba6582) [inline keyboards/dz60/config.h]: I think this should be "DZ" as kbdfans is the vendor not the designer

File: keyboards/dz60/keymaps/via/rules.mk [hunk 2]
```diff
@@ -0,0 +1,3 @@
+MOUSEKEY_ENABLE = no
+LINK_TIME_OPTIMIZATION_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/dz60/keymaps/via/rules.mk]:
```suggestion
LTO_ENABLE = yes
```

Comment (minajevs) [reply keyboards/dz60/keymaps/via/rules.mk]: In [via docs](https://caniusevia.com/docs/configuring_qmk) it says it should be `LINK_TIME_OPTIMIZATION_ENABLE`. I can see there are both `LINK_TIME_OPTIMIZATION_ENABLE` and `LTO_ENABLE` used around the code. Which one should I stick with?

Comment (noroadsleft) [reply keyboards/dz60/keymaps/via/rules.mk]: `LTO_ENABLE` is just an alias for `LINK_TIME_OPTIMIZATION_ENABLE`. The end result is the same.

https://github.com/qmk/qmk_firmware/blob/004ef3fad783832a5b61baa331b45ef5ce67fb9b/tmk_core/common.mk#L206-L215

**Comment by Wilba6582** [comment]
Looks good to me.

Ideally DZ60 should be moved to `keyboards/dztech/` but that could be done after this PR. Perhaps @fauxpark can comment on this.

**Comment by drashna** [comment]
@Wilba6582 the DZ60 doesn't belong to DZTECH, though. It appears to be a KBD Fans board.

File: keyboards/dz60/keymaps/via/rules.mk [hunk 3]
```diff
@@ -0,0 +1,3 @@
+MOUSEKEY_ENABLE = no
```
Comment (noroadsleft) [inline keyboards/dz60/keymaps/via/rules.mk]: I'd like to at least see if this compiles to a workable size in Travis CI without Mouse Keys disabled, as that's a behaviour change from the default keymap.

```suggestion
```

**Comment by minajevs** [comment]
@noroadsleft I've enabled mousekeys back, and VIA builds just fine.

**Comment by mechmerlin** [comment]
LGTM! I'm happy this is going in. A lot of keymap creation questions I get are on a dz60 (though these days starting to see an uptick in dz60rgb-***), but this should decrease those questions.

**Comment by mechmerlin** [comment]
Also, we recently merged in some VIA RGB changes. You may want to add those in on your next PR =).

**Pull request merged**